pkg/controller/operators/catalog/operator.go

```diff
@@ -446,6 +446,12 @@ func (o *Operator) syncCatalogSources(obj interface{}) (syncError error) {
 		o.sourcesLastUpdate = timeNow()
 		logger.Debug("registry server recreated")
 
+		func() {
```

Comment:
this is the fix, everything else is small things I noticed when reviewing

Reply:
Any reason you use a closure instead of just lock/unlock without defer?

```
			o.sourcesLock.Lock()
			delete(o.sources, sourceKey)
			o.sourcesLock.Unlock()
```

Reply:
it's purely to keep the lock/unlock always next to each other, so that in the future if we need to do additional work with `sources` here it's harder to make a mistake

Reply:
Cool, just curious if I could learn something here, thanks.